src/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -1387,6 +1398,82 @@ else if (split[1].equalsIgnoreCase("withdraw")) {
 				TownyMessaging.sendMessage(sender, withdrawMessage);
 				TownyMessaging.sendPrefixedNationMessage(nation, withdrawMessage);
 			}
+			else if (split[1].equalsIgnoreCase("rank")) {
+				if (split.length < 5) {
+					sender.sendMessage(ChatTools.formatTitle("/townyadmin nation rank"));
+					sender.sendMessage(ChatTools.formatCommand(Translation.of("admin_sing"), "/townyadmin nation rank add [resident] [rank] ", ""));
+					sender.sendMessage(ChatTools.formatCommand(Translation.of("admin_sing"), "/townyadmin nation rank remove [resident] [rank] ", ""));
+					return;
+				}
+				Resident target;
+				String rank;
+
+				try {
+					target = getResidentOrThrow(split[3]);
+				} catch (TownyException exception) {
+					TownyMessaging.sendMessage(sender, exception.getMessage());
+					return;
+				}
+				rank = TownyPerms.matchNationRank(split[4]);
+				if (rank == null) {
+					TownyMessaging.sendErrorMsg(player, Translation.of("msg_unknown_rank_available_ranks", split[4], StringMgmt.join(TownyPerms.getNationRanks(), ", ")));
+					return;
+				}
+
+				switch(split[2].toLowerCase()) {
+					case "add":
+						NationRankAddEvent nationRankAddEvent = new NationRankAddEvent(nation, rank, target);
+						BukkitTools.getPluginManager().callEvent(nationRankAddEvent);
+						
+						if (nationRankAddEvent.isCancelled()) {
+							TownyMessaging.sendErrorMsg(player, nationRankAddEvent.getCancelMessage());
+							return;
+						}

```

Comment:
I don't think we want events thrown here, the ta town rank command doesn't. If an admin is executing this they are likely not wanting any other plugin to stop them.